src/bout++.cxx

```diff
@@ -426,136 +415,176 @@ int BoutInitialise(int &argc, char **&argv) {
+    if (output.open("%s/%s.%d", data_dir.c_str(), log_file.c_str(), MYPE)) {
+      throw BoutException(_("Could not open %s/%s.%d for writing"), data_dir.c_str(),
+                          log_file.c_str(), MYPE);
     }
+  }
 
-    /// Add book-keeping variables to the output files
-    bout::globals::dump.add(const_cast<BoutReal&>(BOUT_VERSION), "BOUT_VERSION", false);
-    bout::globals::dump.add(simtime, "t_array", true); // Appends the time of dumps into an array
-    bout::globals::dump.add(iteration, "iteration", false);
+  output_error.enable(verbosity > 0);
+  output_warn.enable(verbosity > 1);
+  output_progress.enable(verbosity > 2);
+  output_info.enable(verbosity > 3);
+  output_verbose.enable(verbosity > 4);
+  // Only actually enabled if also compiled with DEBUG
+  output_debug.enable(verbosity > 5);
 
-    ////////////////////////////////////////////
+  // The backward-compatible output object same as output_progress
+  output.enable(verbosity > 2);
+}
 
-    bout::globals::mesh->outputVars(bout::globals::dump); ///< Save mesh configuration into output file
-    
-  }catch(BoutException &e) {
-    output_error.write(_("Error encountered during initialisation: %s\n"), e.what());
-    throw;
+void setRunStartInfo(Options& options) {
+  auto& runinfo = options["run"];
+
+  // Note: have to force value, since may already be set if a previously
+  // output BOUT.settings file was used as input
+  runinfo["version"].force(BOUT_VERSION_STRING, "");
+  runinfo["revision"].force(BUILDFLAG(REVISION), "");
+
+  time_t start_time = time(nullptr);
+  runinfo["started"].force(ctime(&start_time), "");
+}
+
+void setRunFinishInfo(Options& options) {
+  time_t end_time = time(nullptr);
+  options["run"]["finished"].force(ctime(&end_time), "");
+}
+
+Datafile setupDumpFile(Options& options, Mesh& mesh, const std::string& data_dir) {
+  // Check if restarting
+  const bool append = options["append"].withDefault(false);
+
+  // Get file extensions
+  const auto dump_ext = options["dump_format"].withDefault(std::string{"nc"});
+
+  output_progress << "Setting up output (dump) file\n";
+
+  auto dump_file = Datafile(&(options["output"]), &mesh);
+
+  if (append) {
+    dump_file.opena("%s/BOUT.dmp.%s", data_dir.c_str(), dump_ext.c_str());
+  } else {
+    dump_file.openw("%s/BOUT.dmp.%s", data_dir.c_str(), dump_ext.c_str());
   }
-  
-  return 0;
+
+  // Add book-keeping variables to the output files
+  dump_file.add(const_cast<BoutReal&>(BOUT_VERSION), "BOUT_VERSION", false);
+  // Appends the time of dumps into an array
+  dump_file.add(simtime, "t_array", true);
+  dump_file.add(iteration, "iteration", false);
+
+  // Save mesh configuration into output file
+  mesh.outputVars(dump_file);
+
+  return dump_file;
 }
 
-int bout_run(Solver *solver, rhsfunc physics_run) {
-  
+void writeSettingsFile(Options& options, const std::string& data_dir,
+                       const std::string& settings_file, bool write) {
```

Comment:
Not sure about having this bool which turns off the writing. Why not have at calling site
```
if (write) {
   writeSettingsFile(...)
}
```

Reply:
I think it was because both of the calling sites had the same conditional and I was trying to make things as "flat" as possible.

On reflection, it is a little strange! I'll drop it